client/src/search/Typeahead/Typeahead.js

```diff
@@ -40,60 +53,69 @@ export class Typeahead extends React.Component {
 
     if (search_configs !== current_search_configs) {
       return {
-        matching_results_by_page: [],
+        matching_results: [],
         current_search_configs: search_configs,
       };
     } else {
       return null;
     }
   }
   componentDidUpdate(prevProps, prevState) {
-    const { page_size } = this.props;
-
-    const { query_value, current_search_configs } = this.state;
+    const {
+      query_value,
+      current_search_configs,
+      selection_cursor,
+    } = this.state;
     const {
       query_value: prev_query_value,
       current_search_configs: prev_search_configs,
+      selection_cursor: prev_selection_cursor,
     } = prevState;
 
     if (
       query_value !== prev_query_value ||
       current_search_configs !== prev_search_configs
     ) {
-      const matching_results_by_page = !this.show_menu
+      const matching_results = !this.show_menu
         ? []
-        : _.chain(this.all_options)
-            .filter(({ config_group_index, data }) =>
-              // could use optional chaining, but we WANT this to fail fast and loud, to catch
-              // malformed search_configs during development. Should be safe otherwsie
-              this.config_groups[config_group_index].group_filter(
-                query_value,
-                data
-              )
+        : _.filter(this.all_options, ({ config_group_index, data }) =>
+            // could use optional chaining, but we WANT this to fail fast and loud, to catch
+            // malformed search_configs during development. Should be safe otherwsie
+            this.config_groups[config_group_index].group_filter(
+              query_value,
+              data
             )
-            .chunk(page_size)
-            .value();
+          );
 
       this.setState({
-        matching_results_by_page,
-        pagination_cursor: this.default_pagination_cursor,
+        matching_results,
         selection_cursor: this.default_selection_cursor,
       });
-    } else {
-      const active_item = this.active_item;
+    }
+
+    if (this.list_ref.current) {
+      this.list_ref.current.Grid.measureAllCells();
+      const list_height = this.list_ref.current.Grid.getTotalRowsHeight();
+      if (list_height !== this.state.list_height) {
+        if (list_height < 400) {
+          this.setState({ list_height: list_height });
+        } else if (this.state.list_height < 400) {
+          this.setState({ list_height: 400 });
+        }
+      }
```

Comment:
Ouch, played around a little trying to keep the `height` prop static by moving the visible styles from `typeahead__dropdown` to the virtualized list items... but that creates awkward UX problems where parts of the UI that seem visible are actually covered by the virtualization window still.

Seems this is the best fix! I'd recommend offloading this state/update logic to a wrapped version of the virtualized `List` though. Abstract the weirder details of virtualization away from the `Typeahead`. Something like `VirtualizedListWithMaxHeight` (feel free to try and think of a better name!), would straight pass through all its props to the wrapped virtualized `List`, but it would take a `max_height` instead of a `height` and would do this state management internally.

... well, refs would be a small gotcha with something like that, but I think that would work out if the constructor did something like `this.ref = this.props.ref || React.createRef();`